blockchain/chain_sync/src/sync.rs

```diff
@@ -448,26 +449,24 @@ where
     }
     /// Returns message root CID from bls and secp message contained in the param Block
     fn compute_msg_data(
-        &self,
         bls_msgs: &[UnsignedMessage],
         secp_msgs: &[SignedMessage],
     ) -> Result<Cid, Error> {
+        let temp_store = MemoryDB::default();
         // collect bls and secp cids
         let bls_cids = cids_from_messages(bls_msgs)?;
         let secp_cids = cids_from_messages(secp_msgs)?;
         // generate Amt and batch set message values
-        let bls_root = Amt::new_from_slice(self.chain_store.blockstore(), &bls_cids)?;
-        let secp_root = Amt::new_from_slice(self.chain_store.blockstore(), &secp_cids)?;
+        let bls_root = Amt::new_from_slice(&temp_store, &bls_cids)?;
+        let secp_root = Amt::new_from_slice(&temp_store, &secp_cids)?;
 
         let meta = TxMeta {
             bls_message_root: bls_root,
             secp_message_root: secp_root,
         };
         // TODO this should be memoryDB for temp storage
         // store message roots and receive meta_root
-        let meta_root = self
-            .chain_store
-            .blockstore()
+        let meta_root = temp_store
             .put(&meta, Blake2b256)
             .map_err(|e| Error::Other(e.to_string()))?;
```

Comment:
you're going to need to put this back to what it was (or pass in the blockstore in place of where self was) this will not work. You're querying the messages from an empty temporary store then not putting the metadata in the store (temp storage is dropped at the end of this)